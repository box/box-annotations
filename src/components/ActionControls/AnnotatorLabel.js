// @flow
import React from 'react';
import { injectIntl } from 'react-intl';

import CommentText from '../../../third-party/components/Comment/CommentText';

import { TYPES } from '../../constants';
import { isHighlightAnnotation } from '../../util';

import messages from './messages';

import './ActionControls.scss';

type Props = {
    id: string,
    type: AnnotationType,
    currentUser?: User,
    intl: any
};

class AnnotatorLabel extends React.PureComponent<Props> {
    getAnnotationLabel() {
        const { type, currentUser, intl } = this.props;
        const anonymousUserName = intl.formatMessage(messages.anonymousUserName);
        const annotatorName = !!currentUser && !!currentUser.name ? currentUser.name : anonymousUserName;

        if (isHighlightAnnotation(type)) {
            return intl.formatMessage(messages.whoHighlighted, { name: annotatorName });
        }

        if (type === TYPES.draw) {
            return intl.formatMessage(messages.whoDrew, { name: annotatorName });
        }

        return intl.formatMessage(messages.whoAnnotated, { name: annotatorName });
    }

    render() {
        const { id } = this.props;
        const annotatorLabelMessage = this.getAnnotationLabel();

        return (
            <span className='ba-annotation-label'>
                <CommentText id={id} tagged_message={annotatorLabelMessage} translationEnabled={false} />
            </span>
        );
    }
}

export default injectIntl(AnnotatorLabel);
