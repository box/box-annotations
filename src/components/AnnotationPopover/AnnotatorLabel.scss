@import "../../variables";

.ba {
    .ba-annotator-label {
        margin: 10px 0 10px 10px;
        padding-right: 10px;
        border-right: 1px solid $see-see;
    }

    // MOBILE & TABLET CSS
    @media #{$mobile}, #{$tablet} {
        .ba-annotator-label {
            position: fixed;
            top: 0;
            left: 0;
            display: flex;
            justify-content: center;
            width: 100%;
            padding: 20px;
            background: white;
            border-bottom: 1px solid #ccc;
        }
    }
}
