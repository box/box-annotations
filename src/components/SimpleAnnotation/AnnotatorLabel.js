// @flow
import React from 'react';
import { injectIntl } from 'react-intl';

import CommentText from '../../../third-party/components/Comment/CommentText';

import { TYPES } from '../../constants';
import { isHighlightAnnotation } from '../../util';

import messages from './messages';

// import './AnnotatorLabel.scss';

type Props = {
    id: string,
    type: AnnotationType,
    user?: User,
    intl: any
};

class AnnotatorLabel extends React.PureComponent<Props> {
    getAnnotationLabel() {
        const { type, user, intl } = this.props;
        const anonymousUserName = intl.formatMessage(messages.anonymousUserName);
        const annotatorName = !!user && !!user.name ? user.name : anonymousUserName;

        if (isHighlightAnnotation(type)) {
            return intl.formatMessage(messages.whoHighlighted, { name: annotatorName });
        }

        if (type === TYPES.draw) {
            return intl.formatMessage(messages.whoDrew, { name: annotatorName });
        }

        return intl.formatMessage(messages.whoAnnotated, { name: annotatorName });
    }

    render() {
        const { id } = this.props;
        const annotatorLabelMessage = this.getAnnotationLabel();

        return (
            <span className='ba-annotation-label'>
                <CommentText id={id} tagged_message={annotatorLabelMessage} translationEnabled={false} />
            </span>
        );
    }
}

export default injectIntl(AnnotatorLabel);
