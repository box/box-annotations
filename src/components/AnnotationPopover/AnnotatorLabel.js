// @flow
import React from 'react';
import { injectIntl } from 'react-intl';
import get from 'lodash/get';

import CommentText from '../../../third-party/components/Comment/CommentText';

import { isHighlightAnnotation, isDrawingAnnotation } from '../../util';
import messages from './messages';

import './AnnotatorLabel.scss';

type Props = {
    id?: string,
    type: AnnotationType,
    createdBy: ?User,
    isPending: boolean,
    intl: any
};

class AnnotatorLabel extends React.PureComponent<Props> {
    /**
     * Return a string describes the action completed by the annotator
     * @return {string} Localized annotator label message
     */
    getAnnotatorLabelMessage(): string {
        const { type, createdBy, intl } = this.props;
        const anonymousUserName = intl.formatMessage(messages.anonymousUserName);
        const name = get(createdBy, 'name', anonymousUserName);

        let labelMessage = messages.whoAnnotated;
        if (isHighlightAnnotation(type)) {
            labelMessage = messages.whoHighlighted;
        } else if (isDrawingAnnotation(type)) {
            labelMessage = messages.whoDrew;
        }

        return intl.formatMessage(labelMessage, { name });
    }

    render() {
        const { id, isPending } = this.props;
        return (
            !isPending && (
                <span className='ba-annotator-label'>
                    <CommentText id={id} tagged_message={this.getAnnotatorLabelMessage()} translationEnabled={false} />
                </span>
            )
        );
    }
}

export default injectIntl(AnnotatorLabel);
