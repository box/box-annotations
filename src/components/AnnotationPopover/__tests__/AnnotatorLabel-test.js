import * as React from 'react';
import { shallow } from 'enzyme';

import AnnotatorLabel from '../AnnotatorLabel';

const createdBy = {
    type: 'user',
    id: '789ghi',
    name: 'Mia Thermopolis',
    email: 'princess@genovia.gov',
};

describe('components/AnnotationPopover/AnnotatorLabel', () => {
    const render = (props = {}) => shallow(<AnnotatorLabel id="123" {...props} />).dive();

    test('should render nothing if label is pending', () => {
        const wrapper = render({ isPending: true });
        expect(wrapper).toMatchSnapshot();
    });

    test('should render a highlight annotation label', () => {
        const wrapper = render({
            type: 'highlight',
            createdBy,
        });
        expect(wrapper).toMatchSnapshot();
    });

    test('should render a drawing annotation label', () => {
        const wrapper = render({
            type: 'draw',
            createdBy,
        });
        expect(wrapper).toMatchSnapshot();
    });

    test('should render an anonymous highlight annotation label', () => {
        const wrapper = render({
            type: 'highlight',
        });
        expect(wrapper).toMatchSnapshot();
    });

    test('should render an anonymous drawing annotation label', () => {
        const wrapper = render({
            type: 'draw',
        });
        expect(wrapper).toMatchSnapshot();
    });
});
