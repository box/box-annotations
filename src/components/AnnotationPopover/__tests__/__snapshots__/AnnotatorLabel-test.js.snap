// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AnnotationPopover/AnnotatorLabel should render a drawing annotation label 1`] = `
<AnnotatorLabel
  createdBy={
    Object {
      "email": "princess@genovia.gov",
      "id": "789ghi",
      "name": "Mia Thermopolis",
      "type": "user",
    }
  }
  id="123"
  intl={
    Object {
      "formatDate": [Function],
      "formatMessage": [Function],
    }
  }
  type="draw"
/>
`;

exports[`components/AnnotationPopover/AnnotatorLabel should render a highlight annotation label 1`] = `
<AnnotatorLabel
  createdBy={
    Object {
      "email": "princess@genovia.gov",
      "id": "789ghi",
      "name": "Mia Thermopolis",
      "type": "user",
    }
  }
  id="123"
  intl={
    Object {
      "formatDate": [Function],
      "formatMessage": [Function],
    }
  }
  type="highlight"
/>
`;

exports[`components/AnnotationPopover/AnnotatorLabel should render an anonymous drawing annotation label 1`] = `
<AnnotatorLabel
  id="123"
  intl={
    Object {
      "formatDate": [Function],
      "formatMessage": [Function],
    }
  }
  type="draw"
/>
`;

exports[`components/AnnotationPopover/AnnotatorLabel should render an anonymous highlight annotation label 1`] = `
<AnnotatorLabel
  id="123"
  intl={
    Object {
      "formatDate": [Function],
      "formatMessage": [Function],
    }
  }
  type="highlight"
/>
`;
